lib/post_revisor.rb

```diff
@@ -651,6 +652,14 @@ def update_category_description
     end
   end
 
+  def hide_welcome_topic_banner
+    return false unless guardian.is_admin?
+    return false unless guardian.user.id == User.first_login_admin_id
+    return false unless @topic.id == SiteSetting.welcome_topic_id
+
+    MessageBus.publish("/site/welcome-topic-banner", false)
+  end
```

Comment:
I don't think we need to run this check every time a post is revised. Is the logic to decide whether the welcome topic banner is shown cached somewhere in the database?

Reply:
The logic is defined in [`site.rb`](https://github.com/discourse/discourse/blob/d57bea4de36cb2c936b8bacfb998a6925a1ea672/app/models/site.rb#L206-L211). I'll see if we can access it from `post_revisor.rb`.

Reply:
I think we need a cache for the `Site.show_welcome_topic_banner?` method because it is triggering an extra postgres query for all admins even when the welcome topic banner is never going to be shown.

Reply:
I've made the suggested change.